design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+and updates the YAMLs that are packaged in the stack image according to user's
+input on the CR instance of that stack that user creates.
+
+As sky is the limit when it comes to stacks, we need to put down some general
+rules to be adhered by initial configuration stacks.
+
+* CRDs of the resources that will be deployed by the stack are declared as
+  dependency so that user knows what CR instances will be deployed by the given
+  configuration stack.
+* It should be possible to have different sets of configurations in one stack
+  and user should be able to choose from them via the CR that is an instance
+  of stack's CRD.
+  * Note that it's highly preferred to have 1 custom resource definition that
+    the stack uses and expose the high level variations of the parameters through
+    `spec` instead of having different custom resource definitions for each class
+    of configuration.
+* A minimal high level set of configurations are exposed to the user through
+  the stack's CR, such as `Provider` reference to refer in the deployed resource
+  or `region`.
+* Controller should signal readiness of all resources in CR's status.
+* All resources should be labelled referring to the CR instance of the stack.
+* In YAMLs, resources should refer to each other using cross-resource references
+  wherever possible.
+* **TBD** Controller should update the resources continuously and treat the given CR as
+  the source of truth even though user manually changes the resources that are
+  deployed by the stack.
+* **TBD** CRD of the configuration stack should be namespaced so that different
+  teams/users can create their own configured environment.
+* Stack type will be `ClusterStack` as most of the resources it deploys are
+  cluster-scoped.
+
+Name of the concept is **TBD**; _EasyStack_, _ConfigurationStack_,
+_EnvironmentStack_ are current candidates.
+
+## User Experience
+
+1. Install AWS stack and create your `Provider` object named `aws-provider`.
+2. Install a AWS configuration stack via:
+```yaml
+# exact apiVersion is TBD.
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: easy-aws
+  namespace: crossplane-system
+spec:
+  package: "crossplane/easy-aws:latest"
+```
+
+Now, I want an environment where all my database instances and kubernetes clusters
+are connected to each other in a private VPC, which what this specific configuration
+stack does. Create the following:
+
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+```
+
+Then I wait for status to become ready. After it's done, all resources are deployed.
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
+spec:
+  providerRef:
+    name: aws-provider
+status:
+  state: ready
```

Comment:
Most Crossplane resources today use a conditioned status to represent readiness. Should we do so here?

Reply:
Yes, that makes sense. Per discussion on https://github.com/crossplaneio/crossplane/pull/1110#discussion_r353461481 I think representing a condition for reconciliation status could be enough.